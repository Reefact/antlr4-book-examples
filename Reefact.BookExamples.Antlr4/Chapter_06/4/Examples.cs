#region Usings declarations

using Antlr4.Runtime;

using ApprovalTests;
using ApprovalTests.Reporters;

using Xunit;

#endregion

namespace Reefact.BookExamples.Antlr4.Chapter_06._4 {

    [UseReporter(typeof(VisualStudioReporter))]
    public class Examples {

        [Fact]
        [GraphicalTree("ParseTree1.svg")]
        public void mermaid_style_tree() {
            // Setup
            AntlrInputStream inputStream = AntlrInputStreamReader.Read("t.cymbol", 6, 4);
            GRun             grun        = GRun.Read(inputStream);
            // Exercise
            string mermaidStyleTree = grun.ToMermaidStyleTree();
            // Verify
            Approvals.Verify(mermaidStyleTree);
        }

        [Fact]
        [GraphicalTree("ParseTree2.svg")]
        public void precedence_01() {
            // Setup
            AntlrInputStream inputStream = AntlrInputStreamReader.Read("int a = -x+y;");
            GRun             grun        = GRun.Read(inputStream);
            // Exercise
            string mermaidStyleTree = grun.ToMermaidStyleTree();
            // Verify
            Approvals.Verify(mermaidStyleTree);
        }

        [Fact]
        [GraphicalTree("ParseTree3.svg")]
        public void precedence_02() {
            // Setup
            AntlrInputStream inputStream = AntlrInputStreamReader.Read("float b = -a[i];");
            GRun             grun        = GRun.Read(inputStream);
            // Exercise
            string mermaidStyleTree = grun.ToMermaidStyleTree();
            // Verify
            Approvals.Verify(mermaidStyleTree);
        }

    }

}