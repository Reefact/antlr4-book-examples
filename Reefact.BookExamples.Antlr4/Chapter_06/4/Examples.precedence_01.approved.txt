graph TD
	1["file"] --> 2["varDecl"]
	2 --> 3["type"]
	3 --> 4["int"]
	2 --> 5["a"]
	2 --> 6["="]
	2 --> 7["expr"]
	7 --> 8["expr"]
	8 --> 9["-"]
	8 --> 10["expr"]
	10 --> 11["x"]
	7 --> 12["+"]
	7 --> 13["expr"]
	13 --> 14["y"]
	2 --> 15[";"]

classDef default fill:#fff,stroke:#000,stroke-width:0.25px;
classDef error color:#fff,fill:#FF0000,stroke:#000,stroke-width:0.25px;